src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovyCallSiteSelector.java

```diff
@@ -43,7 +44,27 @@
  */
 class GroovyCallSiteSelector {
 
-    private static boolean matches(@Nonnull Class<?>[] parameterTypes, @Nonnull Object[] parameters) {
+    private static boolean matches(@Nonnull Class<?>[] parameterTypes, @Nonnull Object[] parameters, boolean varargs) {
+        if (varargs) {
+            int fixedLen = parameterTypes.length - 1;
+            Class<?> componentType = parameterTypes[fixedLen].getComponentType();
+            assert componentType != null;
+            int arrayLength = parameters.length - fixedLen;
+            if (arrayLength >= 0) {
+                if (arrayLength == 1 && parameterTypes[fixedLen].isInstance(parameters[fixedLen])) {
+                    // not a varargs call
```

Comment:
To capture reasoning discussed OoB for making this a bug and not a nit:  

> Redefining inputs introduces unexpected behavior that others may not catch without reading the function in entirety and understanding it deeply. 

Also 🐜  that we can remove the initial if case, since it's a no-op.